typing/typecore.ml

```diff
@@ -210,60 +210,79 @@ type tail_or_nontail = Tail | Nontail
 type expected_mode =
   { position : tail_or_nontail;
     escaping_context : escaping_context option;
-    mode : Value_mode.t }
+    mode : Value_mode.t;
+    tuple_modes : Value_mode.t list; }
```

Comment:
I think there's an invariant that for all `t` in `tuple_modes`, `t <= mode` (which would be worth a comment)

Reply:
It's actually the slightly stronger `Value_mode.regional_to_local t <= mode` (or equivalently `t <= regional_to_global mode`). I'll add a comment.